hacep/src/main/java/org/kie/hacep/consumer/CommandHandler.java

```diff
@@ -166,8 +183,36 @@ private List getSerializableItemsByNamedQuery(ListObjectsCommandNamedQuery comma
     public void visit(FactCountCommand command) {
         FactCountMessageImpl msg = new FactCountMessageImpl(command.getId(),
                                                             kieSessionContext.getKieSession().getFactCount());
-        producer.produceSync(config.getKieSessionInfosTopicName(),
+        producer.produceSync(envConfig.getKieSessionInfosTopicName(),
                              command.getId(),
                              msg);
     }
+
+    @Override
+    public void visit(SnapshotOnDemandCommand command) {
+        LocalDateTime lastSnapshotTime = sessionSnapshooter.getLastSnapshotTime();
+        LocalDateTime limitAge = LocalDateTime.now().minusSeconds(envConfig.getMaxSnapshotAge());
+        //if the lastSnapshot time is after the the age we perform a snapshot
+        if (lastSnapshotTime == null) {
+            if (((StatefulKnowledgeSessionImpl) kieSessionContext.getKieSession()).isAlive()) {
```

Comment:
Is this alive check necessary? In which case the session could have been disposed? I believe that if this happen the leader is already broken since no other action could be taken on its session anyway.

Reply:
Added first in the DroolcConsuemrHandler to fix an error when the serialization was called during the stop of the pod, but here isn't needed